src/App.tsx

```diff
@@ -1,73 +1,148 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+  useRef,
+  useEffect,
+  MutableRefObject,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from './types/Person';
+
+const delay = 300;
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const people: Person[] = peopleFromServer;
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownActive, setIsDropdownActive] = useState(true);
+  const [chosenPerson, setChosenPerson] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), [
+    setAppliedQuery,
+    delay,
+  ]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setQuery(value);
+    applyQuery(value);
```

Comment:
The filter should not trigger on spaces only. Consider adding a condition to check if the input is only spaces and avoid applying the query in such cases.